src/app/Main/Body/QueryPage/QueryPage-Container.component.tsx

```diff
@@ -3,39 +3,62 @@ import { useHistory } from "react-router-dom";
 import _ from "lodash";
 import { DataContext } from "../../../Shared/Context/DataContext";
 import { DBentry, DBkey } from "../../../Shared/Model/Isolate.model";
-import { mockDataURL } from "../../../Shared/URLs";
+import { filterURL, mockDataURL } from "../../../Shared/URLs";
 import { QueryPageComponent } from "./QueryPage.component";
 import {
     FilterInterface,
+    FilterType,
 } from "../../../Shared/Model/Filter.model";
-import {
-    FilterContext,
-} from "../../../Shared/Context/FilterContext";
+import { FilterContext } from "../../../Shared/Context/FilterContext";
 import { TableContext } from "../../../Shared/Context/TableContext";
 import { getFilterFromPath } from "../../../Core/getFilterFromPath.service";
 import { generatePathString } from "../../../Core/generatePathString.service";
 import { getFeaturesFromPath } from "../../../Core/getTableFromPath.service";
 
+interface FilterConfigApiInterface {
+    filters: {
+        id: string;
+        name: string;
+        parent: string;
+        values: string[];
+    }[];
+}
+
 export function QueryPageContainerComponent(): JSX.Element {
     const { data, setData } = useContext(DataContext);
     const { filter, setFilter } = useContext(FilterContext);
     const { table, setTable } = useContext(TableContext);
     const history = useHistory();
 
     const BASE_URL: string = mockDataURL;
+    const FILTER_URL: string = filterURL;
 
     const getData = async (): Promise<void> => {
         const r: Response = await fetch(BASE_URL);
         const dataProp: DBentry[] = await r.json();
         const keyValueProps = Object.keys(dataProp[0]) as DBkey[];
 
+        const filterResponse: Response = await fetch(FILTER_URL);
+        const filterProp: FilterConfigApiInterface = await filterResponse.json();
+
+        const mainFilter: FilterType[] = [];
         const uniqueValuesObject: FilterInterface = {};
+        const emptyFilter = {} as FilterInterface;
+
+        filterProp.filters.forEach((filterElement) => {
+            const { name } = filterElement;
+            mainFilter.push(name);
+            uniqueValuesObject[name] = filterElement.values;
+            emptyFilter[name] = [];
+        });
 
-        filter.mainFilter.forEach((filterElement) => {
-            const uniqueValuesPerElement: string[] = _.uniq(
-                _.map(dataProp, filterElement)
-            );
-            uniqueValuesObject[filterElement] = uniqueValuesPerElement;
+        setFilter({

```

Comment:
Is there a reason why you moved this outside the effect? Does it work if you use the browser back and forward buttons with different filter settings?

Reply:
i moved setFilter() inside getData() because auf the API fetch. And getData() is inside the effect, so is setFilter().

Reply:
right, i didn't noticed it is inside getData